src/agent.c

```diff
@@ -815,6 +928,9 @@ int agent_bookkeeping(juice_agent_t *agent, timestamp_t *next_timestamp) {
 				continue;
 
 			if (entry->retransmissions >= 0) {
+                if (entry->sock != INVALID_SOCKET && entry->transport == JUICE_TRANSPORT_TCP)
+                    transport_wait_for_connected(entry->sock, NULL);
```

Comment:
I think blocking the whole thread until the transport connects is not acceptable, it would be better to asynchronously wait for the connection.

Reply:
You are right. I'll think about ways to do it asynchronously.